openid-4-verifiable-presentations-1_0.md

```diff
@@ -1464,6 +1464,180 @@ issuers in Self-Sovereign Identity ecosystems using TRAIN</title>
+
+This profile introduces a new parameter `expected_origins`.
+
+* `expected_origins`: An array of strings, each of the strings representing an origin of the Verifier making the request. This parameter MUST only be used with signed requests. It relates the logical Client Identifier to the physical endpoints that are legit origins for requests on behalf of this Client Identifier and is used to detect request replay.
+
+## Response
+
+Every OID4VP Authorization Request MUST result in a response being provided through the W3C Digital Credentials API. The Authorization Response is a JSON object, where the response parameters as defined for the Response Type are encoded as top-level claims in this JSON object. 
+
+The following is an example of an OID4VP Authorization Response through the API: 
+
+```json
+{
+  "presentation_submission": "...",
+  "vp_token": "..."
+}
+```
+
+Note: All mechanisms for cryptographically protecting the OID4VP response MAY be utilized with the W3C Digital Credentials API, too. 
+
+## Example Flows
+
+As OID4VP is pretty flexible, this section shall shed some light on what flows the authors deem especially useful.  
+
+### Unsigned Request
+
+The Verifier MAY send all the OID4VP request data as JSON elements in the `request` API parameter and reveices the result in the API's `result` parameter. In this case, the Wallet will use the Verifier origin as asserted by the Browser as the Verifers's Client Identifier.  
+
+### Signed Request
+
+The Verifier MAY send a signed request.
+
+The signed request object MAY contain all the parameters listed above except `request`. The signed request object MUST contain an `expected_origins` parameter. 
+
+This is a example of such a request:
+
+```JavaScript
+const credential = await navigator.identity.get({
+  digital: {
+    providers: [{
+      protocol: "urn:openid.net:oid4vp",
+      request: {
+       "client_id": "client.example.org",
+       "client_id_scheme": "entity_id",
+       "request": "eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJ..."
+     }
+    }]
+  }
+});
+```
+
+This is an example signed request payload:
+
+```JSON
+{
+    "client_id": "client.example.org",
+    "client_id_scheme": "entity_id",
+    "expected_origins": [
+        "https://origin1.example.com",
+        "https://origin2.example.com"
+    ],
+    "response_type": "vp_token",
+    "nonce": "n-0S6_WzA2Mj",
+    "client_metadata": {
+        "vp_formats": {
+            "vc+sd-jwt": {
+                "sd-jwt_alg_values": [
+                    "PS256"
+                ],
+                "kb-jwt_alg_values": [
+                    "PS256"
+                ]
+            }
+        },
+        "jwks": {
+            "keys": [
+                {
+                    "kty": "EC",
+                    "crv": "P-256",
+                    "x": "MKBCTNIcKUSDii11ySs3526iDZ8AiTo7Tu6KPAqv7D4",
+                    "y": "4Etl6SRW2YiLUrN5vfvVHuhp7x8PxltmWWlbbM4IFyM",
+                    "use": "enc",
```

Comment:
doesn't the presence of an encryption key here suggest that a jarm response mode should be used?

Reply:
that's a good point - will add a text clarifying that the response is encrypted using JARM. @timcappalli @samuelgoto is there clarity if if encrypting the response sent via the browser API is mandatory?

Reply:
I added text on JARM, please review if what has been added is sufficient.

Reply:
look like it's ok